sherpa/python/sherpa/utils.py

```diff
@@ -0,0 +1,68 @@
+import argparse
+
+
+def add_beam_search_arguments():
```

Comment:
Shall we pass the parser as an argument?

We have been doing this in icefall. Please see
https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/tdnn_lstm_ctc/asr_datamodule.py#L77

```python3
    @classmethod
    def add_arguments(cls, parser: argparse.ArgumentParser):
```

Reply:
ActuaIly, I was following : https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/pruned_transducer_stateless3/train.py#L328

Also, in this case, we return a second parser, one for the server and one for the beam search

Reply:
Either way is fine with me.

Reply:
By the way, one possible issue with the current approach is that it cannot handle unknown arguments.

Imagine that a user misspells an argument or provides an unknown/unsupported argument, the current approach just ignores/discards  these arguments silently, which may cause subtle errors and produce results not wanted by the user

Reply:
Good point. Let me think about handling that case

Reply:
@csukuangfj I think I found a solution (and pushed it)

Reply:
Thanks. Will check it when I come to home.